src/library/scala/collection/immutable/StringLike.scala

```diff
@@ -288,27 +288,35 @@ self =>
    */
   def toBoolean: Boolean = parseBoolean(toString)
   /**
+   * Parse string as a Byte (string must contain only decimal digits and optional leading '-').
    * @throws java.lang.NumberFormatException - If the string does not contain a parsable byte.
    */
   def toByte: Byte       = java.lang.Byte.parseByte(toString)
   /**
+   * Parse string as a Short (string must contain only decimal digits and optional leading '-').
    * @throws java.lang.NumberFormatException - If the string does not contain a parsable short.
    */
   def toShort: Short     = java.lang.Short.parseShort(toString)
   /**
+   * Parse string as an Int (string must contain only decimal digits and optional leading '-').
    * @throws java.lang.NumberFormatException  - If the string does not contain a parsable int.
    */
   def toInt: Int         = java.lang.Integer.parseInt(toString)
   /**
+   * Parse string as a Long (string must contain only decimal digits and optional leading '-').
    * @throws java.lang.NumberFormatException  - If the string does not contain a parsable long.
    */
   def toLong: Long       = java.lang.Long.parseLong(toString)
   /**
-   * @throws java.lang.NumberFormatException - If the string does not contain a parsable float.
+    * Parse string as a Double (surrounding whitespace removed with trim).
```

Comment:
It's a bit of a nitpick, but I think this reads better as `(surrounding whitespace is removed with a trim)`.  Complete sentences FTW?

Also, I think all the instances of `string` in the description of the method are redundant.  Just `Parse as a Long` is enough--it's clear from context that it's talking about the string.  (You need the word `string` in the description of the exceptions.  Those should stay as-is.)